azure/scope/managedmachinepool.go

```diff
@@ -219,6 +219,16 @@ func (s *ManagedMachinePoolScope) DeleteLongRunningOperationState(name, service
 	futures.Delete(s.ControlPlane, name, service)
 }
 
+// UpdateMachinePoolReplicas updates capi machinepool replica count.
+func (s *ManagedMachinePoolScope) UpdateMachinePoolReplicas(replicas *int32) error {
+	patch := client.MergeFrom(s.MachinePool.DeepCopy())
```

Comment:
I'm unable to comment on whether or not this patch implementation is what we want, everything else lgtm in this PR.

Thanks again!

Maybe @CecileRobertMichon can give this a final thumbs up or offer feedback for improvement.